packages/wonder-blocks-data/util/response-cache.js

```diff
@@ -116,21 +126,132 @@ export class ResponseCache {
+    };
+
+    /**
+     * Remove from cache, the entry matching the given handler and options.
+     *
+     * This will, if present therein, remove the value from the custom cache
+     * associated with the handler and the framework in-memory cache.
+     *
+     * Returns true if something was removed from any cache; otherwise, false.
+     */
+    remove = <TOptions, TData>(
+        handler: IRequestHandler<TOptions, TData>,
+        options: TOptions,
+    ): boolean => {
+        const requestType = handler.type;
+
+        // NOTE(somewhatabstract): We could invoke removeAll with a predicate
+        // to match the key of the entry we're removing, but that's an
+        // inefficient way to remove a single item, so let's not do that.
+
+        // We don't use custom caches during SSR.
+        const customCache = Server.isServerSide() ? null : handler.cache;
+        const removedCustom: boolean = !!(
+            customCache && customCache.remove(handler, options)
+        );
+
+        // Get the internal subcache for the handler.
+        const handlerCache = this._cache[requestType];
+        if (!handlerCache) {
+            return removedCustom;
+        }
+
+        // Get the entry.
+        const key = handler.getKey(options);
+        const internalEntry = handlerCache[key];
+        if (internalEntry == null) {
+            return removedCustom;
+        }
+
+        // Delete the entry.
+        delete handlerCache[key];
+        return true;
+    };
+
+    /**
+     * Remove from cache, any entries matching the given handler and predicate.
+     *
+     * This will, if present therein, remove matching values from the custom
+     * cache associated with the handler and the framework in-memory cache.
+     *
+     * It returns a count of all records removed. This is not a count of unique
+     * keys, but of unique entries. So if the same key is removed from both the
+     * framework and custom caches, that will be 2 records removed.
+     */
+    removeAll = <TOptions, TData>(
+        handler: IRequestHandler<TOptions, TData>,
+        predicate?: (
+            key: string,
+            cachedEntry: $ReadOnly<CacheEntry<TData>>,
+        ) => boolean,
+    ): number => {
+        const requestType = handler.type;
+
+        // We don't use custom caches during SSR.
+        const customCache = Server.isServerSide() ? null : handler.cache;
+        const removedCountCustom: number =
+            (customCache && customCache.removeAll(handler, predicate)) || 0;
+
+        // Get the internal subcache for the handler.
+        const handlerCache = this._cache[requestType];
+        if (!handlerCache) {
+            return removedCountCustom;
+        }
+
+        // Apply the predicate to what we have cached.
+        let removedCount = 0;
+        for (const [key, entry] of Object.entries(handlerCache)) {
+            if (
+                typeof predicate !== "function" ||
+                predicate(key, (entry: any))
```

Comment:
I assume we're casting to `any` here because Flow's definition for `Object.entries` is getting in the way of nice typing.

Reply:
Yup